### PR TITLE
BUGFIX: Apply enforced image format conversions to all generated images

### PR DESCRIPTION
**Pull request by mficzel**
Previously the configured image format conversions were only applied when a target size was specified. That made the preview images in the media module still not show thumbnails for tiff images. 

In addition the new configuration is much more generic and is easier to support in future when we may improve the implementation.

This change removes the ConvertImageThumbnail Generator and the Configuration that came with it and introduces a setting `Neos.Media.image.defaultOptions.convertFormats` instead. The setting is a key-value-list that allows to specify which media types shall be converted whenever they are rendered. The key in the new configuration is the the source media-type and the value is the target media-type or file-extension.

By default the setting is empty since all format conversions have to be supported by by the imagine driver and especially ``Gd`` lacks some features. 

With the following setting you setup automatic conversion of tiff images to jpg:
```
Neos:
  Media:
    image:
      defaultOptions:
        # Image formats that shall always be converted regardless wether an adjustment was applied or not
        # The key is the the source media-type and the value is the target media-type or file-extension.
        convertFormats:
          image/tiff: image/jpeg
```

!!! This replaces the implementation for the default conversions in #2389 with a more generic approach wich is non breaking since it was not released yet.

### CODE REVIEW

File: Neos.Media/Classes/Domain/Service/ThumbnailService.php [hunk 1]
```diff
@@ -108,6 +116,27 @@ public function injectThrowableStorage(ThrowableStorageInterface $throwableStora
      */
     public function getThumbnail(AssetInterface $asset, ThumbnailConfiguration $configuration)
     {
+        // Enforce format conversions if needed. This replaces the actual
+        // thumbnail-configuration with one tha enforces the target format
```
Comment (bwaidelich) [inline Neos.Media/Classes/Domain/Service/ThumbnailService.php]: nitpick
```suggestion
        // thumbnail-configuration with one that enforces the target format
```

File: Neos.Media/Classes/Domain/Service/ThumbnailService.php [hunk 2]
```diff
@@ -108,6 +116,27 @@ public function injectThrowableStorage(ThrowableStorageInterface $throwableStora
...
+        if ($configuration->getFormat() === null) {
+            if ($targetFormat = Arrays::getValueByPath($this->formatConversions, $asset->getMediaType())) {
+                if (strpos($targetFormat, '/') > 0) {
```
Comment (bwaidelich) [inline Neos.Media/Classes/Domain/Service/ThumbnailService.php]:
```suggestion
                if (strpos($targetFormat, '/') !== false) {
```

File: Neos.Media/Classes/Domain/Service/ThumbnailService.php [hunk 3]
```diff
@@ -108,6 +116,27 @@ public function injectThrowableStorage(ThrowableStorageInterface $throwableStora
      */
     public function getThumbnail(AssetInterface $asset, ThumbnailConfiguration $configuration)
     {
+        // Enforce format conversions if needed. This replaces the actual
+        // thumbnail-configuration with one tha enforces the target format
+        if ($configuration->getFormat() === null) {
+            if ($targetFormat = Arrays::getValueByPath($this->formatConversions, $asset->getMediaType())) {
```
Comment (bwaidelich) [inline Neos.Media/Classes/Domain/Service/ThumbnailService.php]: the many levels make this a little hard to comprehend. Maybe it makes sense to move the default conversion to a private method and replace this by sth like
```php
if (!$configuration->hasFormat()) {
  $configuration = $this->applyDefaultFormat($configuration);
}
```
?

**Comment by mficzel** [comment]
@gerhard-boden i really see no need for a more detailed conversion at this point since quality, interlacing etc. are configured either with default settings already or at the concrete thumbnail.

I decided to go with an empty list of default conversions for the release as @davidspiola and i discovered that the ``Gd`` driver does not support tiff and webp. So at least for now this is will be an opt in feature.

**Comment by gerhard-boden** [comment]
in my experience gd dies no matter if this conversion is happening or not.

**Comment by gerhard-boden** [comment]
but always safer to make new features completely opt-in anyways

**Pull request merged**